src/main/java/view/impl/UserProductMenu.java

```diff
@@ -1,4 +1,84 @@
 package view.impl;
 
+import model.Product;
+import service.BasketService;
+import service.MessageService;
+import service.ProductService;
+import view.MenuOption;
+
+import java.math.BigDecimal;
+import java.util.Scanner;
+
 public class UserProductMenu {
+
+    Scanner scanner = new Scanner(System.in);
+
+    public void operateWithProductMenu() {
+
+        ProductService productService = new ProductService();
+        BasketService basketService = new BasketService();
+        MessageService messageService = new MessageService();
+        int selectedOption;
+
+        do {
+            selectedOption = selectedOptionFromProductMenu();
+
+            switch (selectedOption) {
+                case 1:
+                    productService.showAllProducts();
+                    break;
+                case 2:
+                    productService.searchProductByName();
+                    break;
+                case 3:
+                    basketService.addProductToTheBasket(enterProduct());
+                    break;
+                case 4:
+//                basketService.checkoutAnOrder();
+                    break;
+                case 5:
+                    basketService.showContentOfBasket();
+                    break;
+                case 6:
+                    System.out.println("Enter message to admin: ");
+                    String messageToAdmin = scanner.next();
+                    messageService.sendMessage(1, messageToAdmin);
+                    break;
+                case 0:
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        while (selectedOption != 0);
```

Comment:
why do while?

Reply:
Can't change it. As I was trying to do it with another loop but can't. As I need user after select some menu - get result - and then product menu was shown again so do while here is working. Please suggest better solution as I can't get it.